From 6294c7ae2496f1d05f2a071a006b09db819160fb Mon Sep 17 00:00:00 2001
From: dianlujitao <dianlujitao@lineageos.org>
Date: Tue, 2 Jun 2020 11:11:39 +0800
Subject: [PATCH 6/8] BT: Prefer LDAC over aptX family

Also rename priority key of aptX adaptive r2 to avoid confusion

Change-Id: I490df42ff1d2dd58ca1d9b4f0867814de2a13caa
---
 res/values/config.xml                               | 9 ++-------
 src/com/android/bluetooth/a2dp/A2dpCodecConfig.java | 3 ++-
 2 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/res/values/config.xml b/res/values/config.xml
index c6310dbd5..649a78cad 100644
--- a/res/values/config.xml
+++ b/res/values/config.xml
@@ -101,17 +101,12 @@
          value should be unique. -->
     <integer name="a2dp_source_codec_priority_sbc">1001</integer>
     <integer name="a2dp_source_codec_priority_aac">2001</integer>
-    <integer name="a2dp_source_codec_priority_ldac">3001</integer>
     <integer name="a2dp_source_codec_priority_aptx">4001</integer>
     <integer name="a2dp_source_codec_priority_aptx_hd">5001</integer>
     <integer name="a2dp_source_codec_priority_aptx_adaptive">6001</integer>
     <integer name="a2dp_source_codec_priority_aptx_tws">7001</integer>
-    <!-- max priority is used to dynamically increase the priority
-         of codecs(mainly Aptx Adaptive) to highest value based
-         on config
-         Value of a2dp_source_codec_priority_max should be set to
-         1000 + priority of codec with highest priority-->
-    <integer name="a2dp_source_codec_priority_max">8001</integer>
+    <integer name="a2dp_source_codec_priority_aptx_adaptive_r2">8001</integer>
+    <integer name="a2dp_source_codec_priority_ldac">9001</integer>
 
     <!-- Package that is responsible for user interaction on pairing request,
          success or cancel.
diff --git a/src/com/android/bluetooth/a2dp/A2dpCodecConfig.java b/src/com/android/bluetooth/a2dp/A2dpCodecConfig.java
index b2593cdd8..0d334aeb7 100644
--- a/src/com/android/bluetooth/a2dp/A2dpCodecConfig.java
+++ b/src/com/android/bluetooth/a2dp/A2dpCodecConfig.java
@@ -218,7 +218,8 @@ class A2dpCodecConfig {
                     if(a2dp_offload_cap.contains("aptxadaptiver2")) {
                         int aptxaa_r2_priority;
                         try {
-                            aptxaa_r2_priority = resources.getInteger(R.integer.a2dp_source_codec_priority_max);
+                            aptxaa_r2_priority = resources.getInteger(
+                                    R.integer.a2dp_source_codec_priority_aptx_adaptive_r2);
                         } catch (NotFoundException e) {
                             aptxaa_r2_priority = value;
                         }
-- 
2.34.0

